addons/interaction/functions/fnc_switchWeaponAttachment.sqf

```diff
@@ -60,28 +68,20 @@ if (_removeOld) then {
         };
 
         _unit addItem _oldAttachment;
-
-        // If adding a new item, let that code handle raising EH
-        // Delete addNew from array, to be able to pass _this to EH
-        if (_this deleteAt 4) exitWith {};
-
-        ["CBA_attachmentSwitched", _this] call CBA_fnc_localEvent;
-    }, [_unit, _oldAttachment, _newAttachment, _currWeaponType, _addNew], 0.3] call CBA_fnc_waitAndExecute;
+    }, [_unit, _oldAttachment, _newAttachment, _currWeaponType], 0.3] call CBA_fnc_waitAndExecute;
 };
 
 if (!_addNew) exitWith {};
 
 [{
-    params ["_unit", "", "_newAttachment"];
+    params ["_unit", "", "_newAttachment", "", "_weapon"];
 
     // Delete weapon from array, to be able to pass _this to EH
-    _unit addWeaponItem [_this deleteAt 4, _newAttachment];
+    _unit addWeaponItem [_weapon, _newAttachment];
 
     [[getText (configFile >> "CfgWeapons" >> _newAttachment >> "picture"), 4], true] call CBA_fnc_notify;
 
     if (_unit == ACE_player) then {
         playSound "click";
     };
-
-    ["CBA_attachmentSwitched", _this] call CBA_fnc_localEvent;
```

Comment:
Is there any particular reason why you reverted these changes? Is that EH only supposed to be raised when you switch an attachment mode?

Reply:
I'm not sure if it would cause any problems, but it didn't seem to fit the event
we're just adding or removing two unrelated optics and not really going through cba system

`"CBA_attachmentSwitched"` would normally always have both old and new strings
but here on add or removal one could be `""`

Reply:
but I'm happy to re-add if you think it could be useful for something

Reply:
Nah, I think for players there's the CBA loadout event that should suffice if you decide to remove an attachment.